microservice/src/main/java/com/skytask/repository/ProductRepository.java

```diff
@@ -0,0 +1,19 @@
+package com.skytask.repository;
+
+import com.skytask.common.Product;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.jpa.repository.query.Procedure;
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+
+@Repository
+public interface ProductRepository extends JpaRepository<Product, Long> {
+
+    @Procedure
+    int insertProduct(String name, String description, Double price, Integer stock);
+
+    @Query(nativeQuery = true, value = "SELECT * from  getproducts()")
+    List<Product> getProducts();
```

Comment:
Find another way to make this query, if you're using spring data use the features it has

Reply:
I used it because @Query comes from data jpa as well as @Procedure, but I wasn't able to make it work with procedure.

Reply:
Try harder. 
Using Spring Data features makes code (most of the times) more reusable and readable.

Reply:
for this, try looking at how to use @NamedStoredProcedureQueries annotation to make the @Procedure work